### DISCUSSION TITLE
Allow totally ommit struct argument as the last argument of function

### DISCUSSION THREAD

**Discussion by pavel-kolesnikov** [vlang/v#11710] [Ideas]
In [Trailing struct literal arguments](https://github.com/vlang/v/blob/master/doc/docs.md#trailing-struct-literal-arguments) para-defaults like-a-varargs trick is described. 

Now, assume all our params in actuall function call are defaults. 
I expect that ommiting arguments from `foo` call will be equal to `foo(Foo{})`:
```v
struct Foo {
	baz string = "cool!"
	bar int = 3
}

fn foo(c Foo) int {
	return c.bar
}

fn main() {
	assert foo() == 3
}
```

Nope, this will not compile:
```
hello.v:10:9: error: too few arguments in call to `foo` (0 instead of 1)
    8 |
    9 | fn main() {
   10 |     assert foo() == 3
      |            ~~~~~
   11 | }
```

Adding empty `{}` to `foo()` will make it compile, though:
```v
fn main() {
	assert foo({}) == 3
}
```

**Proposal**
Allow empty arguemnts if the only argument of the function is struct.

**Comment by dumblob** [comment]
I actually don't think this is a good idea - see https://github.com/vlang/v/pull/11135#issuecomment-924741620 . The reason is the cognitive expectation of *knowing which functions have arguments and which do not* by just reading the call site.
